graylog2-web-interface/src/views/components/datatable/bindings.tsx

```diff
@@ -14,13 +14,47 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+
 import type { VisualizationType } from 'views/types';
 import DataTable from 'views/components/datatable/DataTable';
+import DataTableVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/DataTableVisualizationConfig';
+import type { WidgetConfigFormValues } from 'views/components/aggregationwizard';
 
+type DataTableVisualizationConfigFormValues = {
+  pinned_columns: Array<string>,
+};
 const dataTable: VisualizationType<typeof DataTable.type> = {
   type: DataTable.type,
   displayName: 'Data Table',
   component: DataTable,
+  config: {
+    createConfig: () => ({ pinned_columns: [] }),
+    fromConfig: (config: DataTableVisualizationConfig | undefined) => ({ pinned_columns: config?.pinned_columns.toJS() ?? [] }),
+    toConfig: (formValues: DataTableVisualizationConfigFormValues) => DataTableVisualizationConfig.create(formValues.pinned_columns),
+    fields: [{
+      name: 'pinned_columns',
+      title: 'Pinned Columns',
+      type: 'multi-select',
+      options: ({ formValues }: { formValues: WidgetConfigFormValues }) => {
+        const options = formValues?.groupBy?.groupings.reduce((res, cur) => {
+          if (cur.direction === 'row' && cur?.field?.field) {
+            res.push(cur.field.field);
+          }
+
+          return res;
+        }, []) || [];
+
+        /*
+        if (formValues?.metrics) {
```

Comment:
Is there a reason to keep the commented code?

Reply:
We can uncomment this part if we decide to use metrics for pinning. But I can remove it if you think that it is not worth that

Reply:
I think it is simple enough to reproduce it again if we really want to enable it for metrics.